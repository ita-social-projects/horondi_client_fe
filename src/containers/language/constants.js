export const LANGUAGES_LIST = ['ua', 'en'];
