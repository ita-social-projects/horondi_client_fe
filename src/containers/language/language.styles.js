import { makeStyles } from '@material-ui/core/styles';
import { DEFAULT_LANGUAGE } from '../../configs/index';

export const useStyles = makeStyles((theme) => ({
  root: ({ fromSideBar, language }) => ({
    '& .MuiButtonGroup-root': {
      height: '20px',
      '&:hover': {
        backgroundColor: 'none',
        '& .MuiButtonGroup-groupedOutlinedHorizontal:not(:last-child)': {
          borderRight: `1px solid ${fromSideBar ? '' : 'rgba(254, 254, 254, 0.75)'}`
        },
        '& span': {
          color: fromSideBar ? '' : 'rgba(254, 254, 254, 0.75)'
        }
      },
      '& .MuiButton-root': {
        padding: '5px 8px',
        fontStyle: 'normal',
        fontWeight: 600,
        fontSize: '12px',
        letterSpacing: '0.0015em',
        color: fromSideBar ? theme.palette.textColor : 'rgba(254, 254, 254, 0.75)',
        borderRadius: '0px',
        '& > span': {
          zIndex: -1
        }
      }
    },
    '& .MuiButton-outlined': {
      border: 'none'
    },
    '& .MuiButtonGroup-groupedOutlinedHorizontal:last-child': {
      textDecoration: language === DEFAULT_LANGUAGE ? 'none' : 'underline'
    },
    '& .MuiButtonGroup-groupedOutlinedHorizontal:not(:last-child)': {
      textDecoration: language === DEFAULT_LANGUAGE ? 'underline' : 'none',
      borderRight: `1px solid ${
        fromSideBar ? theme.palette.textColor : 'rgba(254, 254, 254, 0.75)'
      }`
    }
  })
}));
