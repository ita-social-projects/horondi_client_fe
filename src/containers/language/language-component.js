import React from 'react';
import { ButtonGroup, Button } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { useStyles } from './language.styles';
import { setToLocalStorage } from '../../services/local-storage.service';
import { LANGUAGE } from '../../configs';
import { LANGUAGES_LIST } from './constants';

const LanguageComponent = ({ fromSideBar }) => {
  const styles = useStyles({ fromSideBar });
  const { i18n } = useTranslation();

  const handleChange = (e) => {
    const targetValue = e.target.value;
    setToLocalStorage(LANGUAGE, targetValue);
    i18n.changeLanguage(targetValue);
  };

  const mappedLanguages = LANGUAGES_LIST.map((lang) => (
    <Button key={lang} value={lang}>
      {lang}
    </Button>
  ));
  return (
    <div data-cy='language' className={styles.root}>
      <ButtonGroup onClick={handleChange}>{mappedLanguages}</ButtonGroup>
    </div>
  );
};

export default LanguageComponent;
