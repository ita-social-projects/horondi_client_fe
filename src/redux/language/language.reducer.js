import { CHANGE_LANGUAGE } from './language.types';

export const initialState = {
  language: 'ua'
};

const languageReducer = (state = initialState, action = {}) => {
  if (action.type === CHANGE_LANGUAGE) {
    return {
      ...state,
      language: action.payload
    };
  }
  return state;
};

export default languageReducer;
