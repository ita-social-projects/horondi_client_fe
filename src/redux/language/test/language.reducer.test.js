/* eslint-disable no-undef */
import languageReducer, { initialState } from '../language.reducer';
import { changeLanguage } from '../language.actions';

describe('test language reducer', () => {
  test('should return default state', () => {
    expect(languageReducer()).toMatchSnapshot();
    expect(languageReducer()).toEqual({
      language: 'ua'
    });
  });
  test('should change language', () => {
    expect(languageReducer(initialState, changeLanguage(1))).toMatchSnapshot();
    expect(languageReducer(initialState, changeLanguage(1))).toEqual({
      language: 1
    });
  });
});
