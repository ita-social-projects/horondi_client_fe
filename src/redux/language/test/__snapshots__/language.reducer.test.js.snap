// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test language reducer should change language 1`] = `
Object {
  "language": 1,
}
`;

exports[`test language reducer should return default state 1`] = `
Object {
  "language": "ua",
}
`;
